demos/common/cpp/models/src/detection_model_retinaface_pt.cpp

```diff
@@ -0,0 +1,250 @@
+#include <utils/slog.hpp>
+#include "models/detection_model_retinaface_pt.h"
+
+#if !defined(_countof)
+#define _countof(x) (sizeof(x) / sizeof(x[0]))
+#endif
+
+ModelRetinaFacePT::ModelRetinaFacePT(const std::string& modelFileName, float confidenceThreshold, bool useAutoResize, float boxIOUThreshold)
+    : DetectionModel(modelFileName, confidenceThreshold, useAutoResize, {"Face"}),  // Default label is "Face"
+    landmarksNum(0), boxIOUThreshold(boxIOUThreshold) {
+}
+
+void ModelRetinaFacePT::prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNetwork) {
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    slog::info << "Checking that the inputs are as the demo expects" << slog::endl;
+    InferenceEngine::InputsDataMap inputInfo(cnnNetwork.getInputsInfo());
+    if (inputInfo.size() != 1) {
+        throw std::logic_error("This demo accepts networks that have only one input");
+    }
+    InferenceEngine::InputInfo::Ptr& input = inputInfo.begin()->second;
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    input->setPrecision(InferenceEngine::Precision::U8);
+    if (useAutoResize) {
+        input->getPreProcess().setResizeAlgorithm(InferenceEngine::ResizeAlgorithm::RESIZE_BILINEAR);
+        input->getInputData()->setLayout(InferenceEngine::Layout::NHWC);
+    }
+    else {
+        input->getInputData()->setLayout(InferenceEngine::Layout::NCHW);
+    }
+
+    //--- Reading image input parameters
+    imageInputName = inputInfo.begin()->first;
+    const InferenceEngine::TensorDesc& inputDesc = inputInfo.begin()->second->getTensorDesc();
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    landmarksNum = 0;
+
+    std::vector<uint32_t> outputsSizes[OT_MAX];
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(output.second->getDims().size()==4 ? InferenceEngine::Layout::NCHW : InferenceEngine::Layout::CHW);
+        outputsNames.push_back(output.first);
+
+        outputsNames.resize(2);
+        if (output.first.find("bbox") != std::string::npos) {
+            outputsNames[OT_BBOX] = output.first;
+        }
+        else if (output.first.find("cls") != std::string::npos) {
+            outputsNames[OT_SCORES] = output.first;
+        }
+        else if (output.first.find("landmark") != std::string::npos) {
+            outputsNames.resize(std::max(outputsNames.size(), (size_t)OT_LANDMARK + 1));
```

Comment:
It is better to use explicit value `3` or the `OT_MAX` instead of `(size_t)OT_LANDMARK + 1` expression.

Reply:
we should resize array to fit OT_LANDMARK to the last item of array. So, we should resize it to OT_LANDMARK+1. It might look little bit clumsy, but internally correct, as numeric value of OT_LANDMARK or OT_MAX may change in future (if we'll adopt this wrapper to another similar model). I'll leave comment about this